src/navigation/reroute.js

```diff
@@ -126,17 +130,13 @@ export function reroute(pendingPromises = [], eventArguments) {
               pendingPromises.forEach(promise => promise.reject(err));
               throw err;
             })
-            .then(() => finishUpAndReturn(false));
+            .then(finishUpAndReturn);
         });
     });
   }
 
-  function finishUpAndReturn(callEventListeners = true) {
```

Comment:
finishUpAndReturn used to be called in two places, but it's now called in only one. That one place (line 129) doesn't need the event listeners to be called so I removed this argument and the if statement below because they're now dead code